controllers/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -67,6 +99,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, nil
 	}
 
+	if tier.Annotations[toolchainv1alpha1.BundledAnnotationKey] == templates.BundledWithHostOperatorAnnotationValue {
+		if !slices.Contains(r.BundledNsTemplateTierKeys, request.NamespacedName) {
+			// this tier used to be bundled with the operator but it is not anymore.
+			// if it is not used by anything, we are safe to delete it.
+			if used, err := r.isTierUsed(ctx, tier); err != nil {
+				return reconcile.Result{}, err
```

Comment:
I would rather make this as part of the finalizer logic - in other words, the "delete" operation can be called, but the actual removal would be blocked until there is at least one space that uses the tier